include/electionguard/ballot.hpp

```diff
@@ -311,14 +311,15 @@ namespace electionguard
         /// given nonce isn't `None`. Likewise, if a crypto_hash is not provided, it will be derived from
         /// the other fields.
         ///</summary>
-        static std::unique_ptr<CiphertextBallotSelection> make_with_precomputed(
```

Comment:
here and in the core classes below, the main changes are 
- removing the `_with_precomputed` and instead allowing the method parameters to dictate the behavior.
- adding public key K to the precompute methods as it was not there before and tis necessary for hashes (also necessary for #216 )
- prefer passing the `TwoTriplesAndAQuadruple` type instead of forcing every caller to call TwoTriplesAndAQuadruple->get_triple1()` and its siblings
- accepting a `HashPrefix` when necessary